packages/fuels-accounts/src/kms/aws/wallet.rs

```diff
@@ -0,0 +1,256 @@
+        self.client
+            .inner()
+            .sign()
+            .key_id(&self.key_id)
+            .signing_algorithm(SigningAlgorithmSpec::EcdsaSha256)
+            .message_type(MessageType::Digest)
+            .message(Blob::new(message.as_ref().to_vec()))
+            .send()
+            .await
+            .map_err(|e| Error::Other(format!("{}: Signing failed - {}", AWS_KMS_ERROR_PREFIX, e)))?
+            .signature
+            .map(|blob| blob.into_inner())
+            .ok_or_else(|| {
+                Error::Other(format!(
+                    "{}: Empty signature response",
+                    AWS_KMS_ERROR_PREFIX
+                ))
+            })
+    }
+
+    fn normalize_signature(
+        &self,
+        signature_der: &[u8],
+        message: Message,
+    ) -> Result<(K256Signature, RecoveryId)> {
+        let mut sig = K256Signature::from_der(signature_der).map_err(|_| {
+            Error::Other(format!("{}: Invalid DER signature", AWS_KMS_ERROR_PREFIX))
+        })?;
+
+        sig = sig.normalize_s().unwrap_or(sig);
+
+        let recovery_id = self.determine_recovery_id(&sig, message)?;
+        Ok((sig, recovery_id))
+    }
+
+    fn determine_recovery_id(&self, sig: &K256Signature, message: Message) -> Result<RecoveryId> {
+        let recid1 = RecoveryId::new(false, false);
+        let recid2 = RecoveryId::new(true, false);
+
+        let correct_public_key = K256PublicKey::from_public_key_der(&self.public_key)
+            .map_err(|_| {
+                Error::Other(format!(
+                    "{}: Invalid cached public key",
+                    AWS_KMS_ERROR_PREFIX
+                ))
+            })?
+            .into();
+
+        let rec1 = VerifyingKey::recover_from_prehash(&*message, sig, recid1);
+        let rec2 = VerifyingKey::recover_from_prehash(&*message, sig, recid2);
+
+        if rec1.map(|r| r == correct_public_key).unwrap_or(false) {
+            Ok(recid1)
+        } else if rec2.map(|r| r == correct_public_key).unwrap_or(false) {
+            Ok(recid2)
+        } else {
+            Err(Error::Other(format!(
+                "{}: Invalid signature (reduced-x form coordinate)",
+                AWS_KMS_ERROR_PREFIX
+            )))
+        }
+    }
+
+    fn format_fuel_signature(
+        &self,
+        signature: K256Signature,
+        recovery_id: RecoveryId,
+    ) -> Signature {
+        let recovery_byte = recovery_id.is_y_odd() as u8;
+        let mut bytes: [u8; 64] = signature.to_bytes().into();
+        bytes[63] = (recovery_byte << 7) | (bytes[63] & 0x7F);
+        Signature::from_bytes(bytes)
+    }
+
+    pub fn address(&self) -> &Bech32Address {
+        &self.fuel_address
+    }
+}
+
+impl AwsWallet {
+    pub async fn with_kms_key(
+        key_id: impl Into<String>,
+        provider: Option<Provider>,
+    ) -> Result<Self> {
+        let config = AwsConfig::from_environment().await;
```

Comment:
Should be changed so that the credentials can be given via arg. But should fix itself if we abstract the aws client.